compiler/modules/bir/verify.bal

```diff
@@ -65,11 +65,242 @@ class VerifyContext {
     }
 }
 
+type RegSet boolean[];
+type BlockSet boolean[];
+
+// The set of registers that are initialized via a given edge starting at `origin` in to a block
+type RegFlow record {|
+    Label? origin; // origin == () iff entry
+    RegSet regs;
+|};
+
 public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode code) returns Error? {
-    VerifyContext cx = new(mod, defn);
-    foreach BasicBlock b in code.blocks {
-        check verifyBasicBlock(cx, b);
+    VerifyContext vc = new(mod, defn);
+    Position funcPos = defn.position;
+    BasicBlock[] blocks = code.blocks;
```

Comment:
Feels to me like `blocks`, `fwdInDegrees`, `preds` and `blocksFlows` should be collected into a single object that is passed around to all these functions.